pkgs/by-name/vi/violet/package.nix

```
@@ -0,0 +1,30 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "violet";
+  version = "0.4.6";
+
+  src = fetchFromGitHub {
+    owner = "paullouisageneau";
+    repo = "violet";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-a/WRFr6C6MWQBAG0PIDdSRVd4wnQgchPeTMoxUa2Qus=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ cmake ];
+
```

Some useful add-ons:

```suggestion

  passthru = {
    updateScript = nix-update-script { };
    tests = testers.testVersion {
      package = violet;
    };
  };

```